delta/plugins/statistics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/statistics/indexing/StatisticsIndexingStream.scala

```diff
@@ -60,14 +85,20 @@ final class StatisticsIndexingStream(
             // Creates a JsonLdPathValueCollection from the event exchange response
             fromEventExchange(view.projectRef, eventExchangeValue)
           }
-          // TODO: Every time a NEW resource (rev = 1) is created, we also need to run the update-by-query-creation task
-          // TODO: Every time an existing resource (rev > 1) is updated and the types change, we also need to run the update-by-update task
           .evalMapFilterValue { res =>
-            res.index(index, includeMetadata = false, sourceAsText = false)
+            res
+              .index(index, includeMetadata = false, sourceAsText = false)
+              .map(_.map(bulk => (res.resource.id, res.resource.types, bulk)))
           }
-          .runAsyncUnit { bulk =>
-            // Pushes INDEX/DELETE Elasticsearch bulk operations
-            IO.when(bulk.nonEmpty)(client.bulk(bulk))
+          .runAsyncUnit { list =>
+            IO.when(list.nonEmpty) {
+              // Performs an update by query for each resource sequentially
+              Task.traverse(list) { case (id, types, _) =>
+                client.updateByQuery(relationshipsQuery(id, types), Set(index.value))
```

Comment:
I don't know if it feasible with painless but is it possible instead of passing a pair of (id, types) for a single resource, to pass a map with all the (id, types) pairs in the current list ?
In order to a single request to ES ?
And the painless script to update the relations from this map and not a simple pair ?

Reply:
I tried that but in the end I didn't go that direction because of the complexity of the final script. It could also  be counter productive having to go through this map of entries for every query result. I expect most of the resources to return nothing or very few resources from the query block on the `_update_by_query`